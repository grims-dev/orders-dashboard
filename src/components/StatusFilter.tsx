import React from "react";
import styled from "styled-components";
const StatusFiltersStyled = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  margin: 30px 0 40px;

  .status {
    all: unset;
    padding: 5px;
    margin: 0 15px;
    border-radius: 10px;
    cursor: pointer;
    text-transform: uppercase;
    font-size: 12px;
    letter-spacing: 2px;
    line-height: 20px;
    transition: 0.1s ease;
    border: 1px solid rgba(255, 255, 255, 0);
    user-select: none;

    &:hover {
      background: rgba(255, 255, 255, 0.1);
    }

    &:focus {
      border-color: rgba(255, 255, 255, 0.3);
    }

    &:last-child {
      margin-right: 0;
    }

    &::before {
      content: "";
      display: inline-block;
      border: 1px solid #fefefe;
      height: 20px;
      width: 20px;
      border-radius: 100%;
      margin-right: 8px;
      vertical-align: bottom;
    }

    &.readytotry {
      &::before {
        background: var(--status-readytotry);
      }
    }
    &.ontheway {
      &::before {
        background: var(--status-ontheway);
      }
    }
    &.inthequeue {
      &::before {
        background: var(--status-inthequeue);
      }
    }
    &.outofstock {
      &::before {
        background: var(--status-outofstock);
      }
    }

    &.status-selected {
      border-color: #fff;
    }
  }
`;

export interface StatusFilterUI {
  handleStatusSelect: (e: any) => void;
  selectedFilter: string;
}

const StatusFilter = (props: StatusFilterUI) => {
  return (
    <StatusFiltersStyled data-test="statusfilter">
      <button
        data-status="readytotry"
        className={`status readytotry ${
          props.selectedFilter === "readytotry" && "status-selected"
        }`}
        onClick={(e) => props.handleStatusSelect(e)}
      >
        Ready to Try
      </button>

      <button
        data-status="ontheway"
        className={`status ontheway ${
          props.selectedFilter === "ontheway" && "status-selected"
        }`}
        onClick={(e) => props.handleStatusSelect(e)}
      >
        On the Way
      </button>

      <button
        data-status="inthequeue"
        className={`status inthequeue ${
          props.selectedFilter === "inthequeue" && "status-selected"
        }`}
        onClick={(e) => props.handleStatusSelect(e)}
      >
        In the Queue
      </button>

      <button
        data-status="outofstock"
        className={`status outofstock ${
          props.selectedFilter === "outofstock" && "status-selected"
        }`}
        onClick={(e) => props.handleStatusSelect(e)}
      >
        Out of Stock
      </button>
    </StatusFiltersStyled>
  );
};

export default StatusFilter;
