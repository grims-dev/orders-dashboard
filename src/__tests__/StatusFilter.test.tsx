import { mount } from "enzyme";
import * as React from "react";
import toJSON from "enzyme-to-json";
import StatusFilter, { StatusFilterUI } from "../components/StatusFilter";

function mountStatusFilter(props: Partial<StatusFilterUI> = {}) {
  const defaultProps: StatusFilterUI = {
    handleStatusSelect: jest.fn(),
    selectedFilter: "readytotry",
  };

  return mount(<StatusFilter {...defaultProps} />);
}

describe("Status Filter renders successfully", () => {
  it("renders without crashing", () => {
    const wrapper = mountStatusFilter();
    const component = wrapper.find('[data-test="statusfilter"]');
    expect(toJSON(component)).toMatchSnapshot();
  });

  it("renders snapshot of Status Filter", () => {
    const wrapper = mountStatusFilter();
    expect(wrapper).toMatchSnapshot();
  });
});
