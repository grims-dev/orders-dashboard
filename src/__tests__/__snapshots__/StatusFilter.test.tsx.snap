// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Filter renders successfully renders snapshot of Status Filter 1`] = `
<StatusFilter
  handleStatusSelect={[MockFunction]}
  selectedFilter="readytotry"
>
  <styled.div
    data-test="statusfilter"
  >
    <div
      className="sc-AxjAm rSFOJ"
      data-test="statusfilter"
    >
      <button
        className="status readytotry status-selected"
        data-status="readytotry"
        onClick={[Function]}
      >
        Ready to Try
      </button>
      <button
        className="status ontheway false"
        data-status="ontheway"
        onClick={[Function]}
      >
        On the Way
      </button>
      <button
        className="status inthequeue false"
        data-status="inthequeue"
        onClick={[Function]}
      >
        In the Queue
      </button>
      <button
        className="status outofstock false"
        data-status="outofstock"
        onClick={[Function]}
      >
        Out of Stock
      </button>
    </div>
  </styled.div>
</StatusFilter>
`;

exports[`Status Filter renders successfully renders without crashing 1`] = `
Array [
  <styled.div
    data-test="statusfilter"
  >
    <div
      className="sc-AxjAm rSFOJ"
      data-test="statusfilter"
    >
      <button
        className="status readytotry status-selected"
        data-status="readytotry"
        onClick={[Function]}
      >
        Ready to Try
      </button>
      <button
        className="status ontheway false"
        data-status="ontheway"
        onClick={[Function]}
      >
        On the Way
      </button>
      <button
        className="status inthequeue false"
        data-status="inthequeue"
        onClick={[Function]}
      >
        In the Queue
      </button>
      <button
        className="status outofstock false"
        data-status="outofstock"
        onClick={[Function]}
      >
        Out of Stock
      </button>
    </div>
  </styled.div>,
  <div
    className="sc-AxjAm rSFOJ"
    data-test="statusfilter"
  >
    <button
      className="status readytotry status-selected"
      data-status="readytotry"
      onClick={[Function]}
    >
      Ready to Try
    </button>
    <button
      className="status ontheway false"
      data-status="ontheway"
      onClick={[Function]}
    >
      On the Way
    </button>
    <button
      className="status inthequeue false"
      data-status="inthequeue"
      onClick={[Function]}
    >
      In the Queue
    </button>
    <button
      className="status outofstock false"
      data-status="outofstock"
      onClick={[Function]}
    >
      Out of Stock
    </button>
  </div>,
]
`;
